### PR TITLE
Add separate "add test" manifest actions for XCTest and Swift Testing

### PR DESCRIPTION
**Pull request by DougGregor**
Builds on https://github.com/apple/sourcekit-lsp/pull/1193 and https://github.com/apple/swift-package-manager/pull/7481

### CODE REVIEW

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test

File: Sources/SourceKitLSP/Swift/CodeActions/PackageManifestEdits.swift [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+        let edits = try AddTarget.addTarget(
+          target,
+          to: scope.file,
+          configuration: .init(testHarness: testingLibrary)
+        )
+
+        actions.append(
+          CodeAction(
+            title: "Add test target (\(libraryName))",
```
Comment (grynspan) [inline Sources/SourceKitLSP/Swift/CodeActions/PackageManifestEdits.swift:80]: A single test target can contain code for both XCTest and swift-testing. (See my comments over on https://github.com/apple/swift-package-manager/pull/7481.)

Comment (DougGregor) [reply Sources/SourceKitLSP/Swift/CodeActions/PackageManifestEdits.swift:80]: As noted over in https://github.com/apple/swift-package-manager/pull/7481, a code action that creates a brand-new test target that uses two different test libraries is not useful.

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test macOS

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
apple/swift-package-manager#7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7509

@swift-ci clean test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci please clean test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift-package-manager/pull/7481

@swift-ci clean test Windows

**Pull request merged**